tests/test_game.py

```diff
@@ -0,0 +1,44 @@
+from unittest import TestCase
+
+from card import Card
+from game import Game
+
+
+class CardTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.game = Game()
+
+    def test_game_deal_hands(self):
+        self.game.deal_game()
+        player_hand_count = len(self.game.player.hand)
+        dealer_hand_count = len(self.game.dealer.hand)
+        self.assertEqual(player_hand_count, 2)
+        self.assertEqual(dealer_hand_count, 2)
+
+    def test_game_discard_hands(self):
+        self.game.discard_hands()
+        player_hand_count = len(self.game.player.hand)
+        dealer_hand_count = len(self.game.dealer.hand)
+        self.assertEqual(player_hand_count, 0)
+        self.assertEqual(dealer_hand_count, 0)
+
+    def test_game_dealer_turn_5(self):
+        self.game.dealer.hand = [Card(2, 1), Card(3, 2)]
+        self.game.turn_dealer()
+        self.assertGreater(self.game.dealer.calculate_score(), 5)
```

Comment:
We can know what is going to happen in `turn_dealer`. Either by seeding the RNG (`random.seed(1)` will make this test the do the same thing everytime). Or we can do some mocking to make the dealer get specific cards. Either way, we'll know what the score is actually supposed to be.

Reply:
Ooh, that was an oversight. I believe it works now with just the `__hash__`. I checked it by adding a duplicate of the first card in the deck after creation - It did fail.

I'm not sure if the way that I built the `__hash__` is best though. I'm familiar with turning ints into strings from doing riddles and stuff, but there could be a more readable way.

Reply:
I think this test is weird to create because the method is poor...? I could seed the random, but would it maybe be better to just fix the structure before going deeper with building these particular tests?

Reply:
Having tests will help make sure when we refactor the code that they are not broken. To note, these tests being hard to write is usually an indication that the code we are testing could use a good refactor.

Reply:
Am I really testing the dealer's score in this test or am I testing that it did indeed 'decide' to take another card? Could I just assert that the hand size increased, or I'm I thinking about it incorrectly?

Reply:
It makes more sense to test that the dealers hand has changed I think. Testing the count would ideally happen elsewhere

Reply:
Isn't testing hand scoring happening elsewhere too, though? In my mind, this test is testing the dealer's 'logic. So, I was just looking for the simplest way to test that the dealer correctly chose to 'hit or 'stay'. It's super possible that I'm not thinking about it correctly.

Reply:
Checking the length of the dealers hand will tell us how many cards he chose. That's probably what we should be using here